### PR TITLE
Update bond trading tests

### LINKED ISSUE #16411: Convert bond-trading tests over to Scripts

### PR DESCRIPTION
**Pull request by samuel-williams-da**
Resolves #16411 

Update the bond-trading tests to use Script over Scenario

Note, JQ queries can no longer run as the parse depth of the output file is too large. Consider switching to `jj` (separate ticket)

### CODE REVIEW

File: compiler/damlc/tests/bond-trading/Setup.daml [hunk 1]
```diff
@@ -76,46 +81,46 @@ template HelperSetupJob
...
+setupExample = script do
+    setDate 2018 May 14
 
-    acme <- getParty "Acme Bank"
+    acme <- allocateParty "Acme Bank"
 
-    job <- marketSetupJobExample
-    submit acme do create job
+    job <- marketSetupJobExample acme
+    submit acme do createCmd job
```
Comment (dylant-da) [inline compiler/damlc/tests/bond-trading/Setup.daml:104]: All other commands were rewritten with inline `submit`, should we change this one too?

**Review by dylant-da — APPROVED** [review]
Thanks for getting this through! Just one tiny nit, feel free to ignore if you think it's too small to bother with waiting for CI just for that.

I'm surprised the JQ is having issues, does its streaming mode work?

File: compiler/damlc/tests/bond-trading/Setup.daml (hunk 1 above)
Comment (samuel-williams-da) [reply compiler/damlc/tests/bond-trading/Setup.daml:104]: Ah oops, I must have missed this

**Comment by samuel-williams-da** [comment]
Wasn't aware jq had a streaming mode, I can give it a go

File: compiler/damlc/tests/bond-trading/Setup.daml [hunk 2]
```diff
@@ -76,46 +81,46 @@ template HelperSetupJob
...
 
-    job <- marketSetupJobExample
-    jobCid <- submit acme do create job
+    job <- marketSetupJobExample acme
+    jobCid <- acme `submit` createCmd job
 
-    rs <- submit acme do exercise jobCid Process
+    rs <- acme `submit` exerciseCmd jobCid Process
 
     forA rs $ \r -> submit r.party do
```
Comment (dylant-da) [inline compiler/damlc/tests/bond-trading/Setup.daml:118]: There's a chance for inline submit here too, but this line hasn't changed so maybe not

**Review by dylant-da — APPROVED** [review]

File: compiler/damlc/tests/bond-trading/Setup.daml (hunk 2 above)
Comment (samuel-williams-da) [reply compiler/damlc/tests/bond-trading/Setup.daml:118]: Yeah this one I was hesitant about, given the complexity of the RHS

**Comment by akrmn** [comment]
> JQ queries can no longer run as the parse depth of the output file is too large

oh wow!

**Comment by samuel-williams-da** [comment]
Yeah, the bond-trading test is a pretty large test, ends up with a tree depth >256, which is jq's limit

**Comment by samuel-williams-da** [comment]
One option might be to split it up into several tests

**Review by akrmn — APPROVED** [review]
Thanks @samuel-williams-da!

I'm looking forward to your experiments with jq streaming mode :)

just one small thing, I think you should now be able to remove this line:

https://github.com/digital-asset/daml/blob/e82a2042dd405f42448510ad73d741b3474ed3d5/compiler/damlc/tests/BUILD.bazel#L755

**Comment by samuel-williams-da** [comment]
Oh good shout!

**Comment by samuel-williams-da** [comment]
Will resolve jq issue in #16447 
Side-note, I dumped out the json file for bond-trading, and its 200k lines :grimacing:

**Pull request merged**